libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI+Files.m

```diff
@@ -103,7 +99,18 @@ - (SFRestRequest *) requestForFileShares:(NSString *)sfdcId page:(NSUInteger)pag
 - (SFRestRequest *) requestForAddFileShare:(NSString *)fileId entityId:(NSString *)entityId shareType:(NSString*)shareType {
     NSString *path = [NSString stringWithFormat:@"/%@/sobjects/ContentDocumentLink", self.apiVersion];
     NSDictionary *params = @{CONTENT_DOCUMENT_ID: fileId, LINKED_ENTITY_ID: entityId, SHARE_TYPE: shareType};
-    return [SFRestRequest requestWithMethod:SFRestMethodPOST path:path queryParams:params];
+    SFRestRequest *request = [SFRestRequest requestWithMethod:SFRestMethodPOST path:path queryParams:nil];
+    NSError *error = nil;
+    if (params) {
```

Comment:
No more trickery of passing `body` for `POST` as `queryParams`. We set the body directly for `POST` and use `queryParams` only for other requests.